src/Spryker/Zed/Sales/Persistence/SalesRepository.php

```diff
@@ -33,4 +35,32 @@ public function findCustomerOrderIdByOrderReference(string $customerReference, s
 
         return $idSalesOrder;
     }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderListTransfer $orderListTransfer
+     *
+     * @return \Generated\Shared\Transfer\OrderListTransfer
+     */
+    public function getCustomerOrderListByCustomerReference(OrderListTransfer $orderListTransfer): OrderListTransfer
+    {
+        $orderListQuery = $this->getFactory()
+            ->createSalesOrderQuery()
+            ->filterByCustomerReference($orderListTransfer->getCustomerReference());
+
+        $ordersCount = $orderListQuery->count();
+        if (!$ordersCount) {
+            return $orderListTransfer;
```

Comment:
does this mean we always expect empty transfer?

I could have passed some data in the transfer object, why we don't set it to the correct value?